io/src/main/scala/sbt/nio/Glob.scala

```diff
@@ -0,0 +1,518 @@
+/*
+ * sbt IO
+ *
+ * Copyright 2011 - 2019, Lightbend, Inc.
+ * Copyright 2008 - 2010, Mark Harrah
+ *
+ * Licensed under Apache License 2.0
+ * (http://www.apache.org/licenses/LICENSE-2.0).
+ */
+
+package sbt.nio
+
+import java.io.{ File, IOException }
+import java.nio.file._
+import java.util
+
+import sbt.io.{ FileFilter, PathFinder, SimpleFileFilter }
+
+import scala.annotation.tailrec
+import scala.collection.mutable.ArrayBuffer
+
+/**
+ * Represents a filtered subtree of the file system.
+ */
+sealed trait Glob {
+
+  /**
+   * The root of the file system subtree.
+   */
+  def base: Path
+
+  /**
+   * Controls which paths should be considered part of the glob based on the number of components
+   * in the path when it is relativized with respect to the base path. The boundaries are inclusive.
+   * A range of `(0, 0)` implies that only the base path is accepted. A range of `(0, 1)` implies
+   * that the base path and its immediate children are accepted but no children of subdirectories
+   * are included. A range of `(1, Int.MaxValue)` implies that all children of the base path are
+   * accepted and so on.
+   *
+   * @return the range of relative path name lengths to accepts.
+   */
+  def range: (Int, Int)
+
+  /**
+   * The filter to apply to elements found in the file system subtree.
+   * @return the filter.
+   */
+  def pathFilter: PathFilter
+
+}
+private[sbt] sealed trait GlobBuilder[G] extends Any {
+  def /(component: String): G
+  def \(component: String): G
```

Comment:
The idea is that on windows the path separator is `\` rather than `/` like on osx and windows. This trait is confusing and I was only using it to convert files to globs (at one point I was also using it to convert Paths to Globs) so I went ahead and just made a single concrete class FileGlobBuilder with `def \(component: String): Glob` and `def \(component: String): Glob`. The reason for FileGlobBuilder is to enable the PathFinder dsl to be used to specify globs. I'm increasingly unsure if that is a good idea or not. Perhaps we should mix and match the apis.

Reply:
Yea. Path on sbt should be expressed using `/` only.

Reply:
Ok, for legacy reasons, it needs to persist in the file based apis for binary compatibility, but I'll remove the `\` methods from GlobOps.